lib/utility/src/WatchdogWrapper.cpp

```diff
@@ -0,0 +1,35 @@
+
+#include "WatchdogWrapper.h"
+
+#include <string>
+
+#include "Logger.h"
+#include "ResetReason.h"
+#include "mbed.h"
+
+namespace Utility {
+
+void WatchdogWrapper::startWatchdog(std::chrono::milliseconds countdown_ms /*= 5000ms*/,
+                                    std::chrono::milliseconds pet_ms /*= 1000ms*/) {
+  uint32_t countdown_uint32 = countdown_ms.count();
+  Watchdog &watchdog        = Watchdog::get_instance();
+  watchdog.start(countdown_uint32);
+  Thread pet_thread;
+  pet_thread.start(callback(WatchdogWrapper::petWatchdog, &pet_ms));
+}
+
+void WatchdogWrapper::logResetReason() {
+  const reset_reason_t reason = ResetReason::get();
+  if (reason == RESET_REASON_WATCHDOG) {
+    time_t seconds = time(NULL);
```

Comment:
Ideally this would be an accurate time and date, but since that's not possible without some idea of what the time and date is when it is run we'll use the time since the microcontroller was powered on.